### ISSUE TITLE
Normalize directive doesn't work for value variables

### ISSUE THREAD

**Issue by peterstace** [dgraph-io/dgraph#1836]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?

master at 8e77bf90

- What is the hardware spec (RAM, OS)?

N/A

- Steps to reproduce the issue (command/config used to run Dgraph).

With an empty db (no data, no schema), run the following query:

```
{
    q(func: uid(0x1)) @normalize {
        one as math(1)
        uno: val(one)
    }
}
```

- Expected behaviour and actual result.

Actual:
```
{
  "q": [
    {
      "val(one)": 1,
      "uno": 1
    }
  ]
}
```

Expected:
```
{
  "q": [
    {
      "uno": 1
    }
  ]
}
```

**Comment by peterstace** [comment]
Some other queries that don't quite work as expected:

- `{ q(func: uid(0x1)) @normalize { uid } }` gives `{"q":[{"uid":"0x1"}]}` but should give `{"q":[{}]}`.
- `{ q(func: uid(0x1)) @normalize { count(uid) } }` gives `{"q":[{"count":1}]}` but should give `{"q":[{}]}`.
- 

And the bizarre:
```
{
    q(func: uid(0x1)) @normalize
    {
        c: count(uid)
        one as math(1)
        uno: val(one)
    }
}
```
```
{
  "q": [
    {
      "c": 1
    },
    {
      "c": 1,
      "uno": 1
    }
  ]
}
```

**Comment by peterstace** [comment]
Fixed at 5489d32
